src/charts/legend.coffee

```diff
@@ -192,45 +195,46 @@ Ember.Charts.Legend = Ember.Mixin.create
   .volatile()
 
   drawLegend: ->
-    @clearLegend()
-    legend = @get 'legend'
-
-    legend.attr(@get 'legendAttrs')
-
-    showLegendDetails = @get 'showLegendDetails'
-    hideLegendDetails = @get 'hideLegendDetails'
-    legendItems = legend.selectAll('.legend-item')
-      .data(@get 'legendItems')
-      .enter().append('g')
-      .attr(@get 'legendItemAttrs')
-      .on("mouseover", (d,i) -> showLegendDetails(d,i,this))
-      .on("mouseout", (d,i) -> hideLegendDetails(d,i,this))
-
-    legendIconAttrs = @get 'legendIconAttrs'
-    isShowingTotal = @get 'isShowingTotal'
-    totalPointShape = @get 'totalPointShape'
-    legendItems.each (d, i) ->
-      sel = d3.select(this)
-      if (i is 0) and isShowingTotal
-        sel.append('g')
-          .attr('class', 'icon')
-          .call(totalPointShape)
-      else
-        d3.select(this).append('path')
-          .attr('class', 'icon')
-          .attr(legendIconAttrs)
-
-    legendLabelWidth = @get 'legendLabelWidth'
-    labelTrimmer = Ember.Charts.Helpers.LabelTrimmer.create
-      getLabelSize: (d) -> legendLabelWidth
-      getLabelText: (d) -> d.label
-
-    labels = legendItems.append('text')
-      .style('text-anchor', 'start')
-      .text((d) -> d.label)
-      .attr(@get 'legendLabelAttrs')
-      .call(labelTrimmer.get 'trim')
-    totalLabelWidth = 0
-    legend.selectAll('text').each ->
-      totalLabelWidth += this.getComputedTextLength()
-    @set 'averageLegendLabelWidth', totalLabelWidth / @get 'legendItems.length'
+    if @get 'showLegend'
```

Comment:
Could we do something like
```if not @get 'showLegend'
    return
```
instead? Fewer levels of indentation = easier to read :P